Feat/sc 78 context page

davidgilbertson opened the pull request:
Jira: https://govausites.atlassian.net/browse/SC-78

This adds 'context' as a new category. The context page is special in a few ways. Each 'sub category' background colour isn't just a different tint of the base colour, it's a specific shade from another category. So I've refactored the code to use explicit colours rather than define only the shade/tint.

I could rip the whole `colorName` thing out, since it always matches the ID, but this would probably cause nasty conflicts with SC-16 so I'll do it later.

Desktop:
![image](https://user-images.githubusercontent.com/4443482/31701891-1a4d98d4-b41f-11e7-8e12-2baf2e6bf790.png)

Commit:
WIP

Commit:
Add colours

Commit:
Tidy up

davidgilbertson commented on src/components/SubCategorySummary/SubCategorySummary.scss:
```diff
@@ -1,5 +1,9 @@
 @import '~style/variables';
 
+.wrapper {
+  border-top: 4px solid $color-white;
```
No strictly related, but added the little white gap that was added to the designs at some point.

micapam commented on src/components/PageBanner/PageBanner.js:
```diff
@@ -38,16 +38,18 @@ const PageBanner = (props) => {
-            isCategoryPage={props.isCategoryPage}
-            size="LARGE"
-          />
+        {!props.isContextPage && (
```
Ah is this the usual syntax? I had assumed `boolean || <markup />`

davidgilbertson replied:
I've only ever seen the && syntax used, so it reads more like an `if`. When I saw your `||` I almost commented but figured it doesn't really make any difference.

Commit:
Merge remote-tracking branch 'origin/master' into feat/SC-78-context-page

# Conflicts:
#	src/components/AllCitiesCategoryContainer/AllCitiesCategory/AllCitiesCategory.js
#	src/components/AllCitiesCategoryContainer/AllCitiesCategory/__snapshots__/AllCitiesCategory.test.js.snap
#	src/components/SubCategorySummary/SubCategorySummary.js
#	src/constants/categories.js
#	src/constants/indicators.js

Commit:
Post-merge tidy up

micapam reviewed (changes requested):
Not sure what's causing this, but I notice that, in desktop view (n.b. on my small laptop, its screen seems to be - or emulate, since it's retina - 1275px in width), when I'm looking at a category, I can't see the 'Overview' button any more in the category nav - it gets tucked away behind the cities nav. It may well be this is not a new bug per se but just something that has been exposed by the change in elements on the screen:

![image](https://user-images.githubusercontent.com/165087/31801727-dbee450c-b595-11e7-82a1-aa91a7104eda.png)

Also, in mobile view, if I'm on a category midway through the list, so that the category nav scrolls to position, I can't see the little edge of the button off to the left any more, to show me that there are other categories offscreen ...

<img width="361" alt="missing_button" src="https://user-images.githubusercontent.com/165087/31801860-8b445aa0-b596-11e7-8f24-85e66786b85e.png">

Commit:
Fix navigation padding issue

davidgilbertson commented:
Ah, the nav items were too wide for the screen at just over 1200px. So when you clicked a category, they would get scrolled to the right a bit and the left one hidden. I've fixed that with slightly smaller spacing for desktop, then large again for 'big desktop' (1440px+)

There was a second problem of padding on the left which I introduced last week some time. I've fixed that too.

The width at 1200px now:

![image](https://user-images.githubusercontent.com/4443482/31804096-f1c91390-b5a2-11e7-893f-51760d35abd1.png)

micapam commented:
Mobile & desktop views both look good now. lgtm
